web/modules/custom/form_api/src/UserFormValidator.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Drupal\form_api;
+
+use Drupal\Component\Utility\EmailValidatorInterface;
+use Drupal\Core\Ajax\AddCssCommand;
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\HtmlCommand;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a service to validate user's form.
+ */
+class UserFormValidator {
```

Comment:
Why create a service instead of defining the methods in the form class itself?

Reply:
Sir isn't it better this way? As I was learning about separation of concerns, the rendering and submitting of form is done using the form class, so I thought to validate the form using this service which also includes a service to validate the email. Should I change it and do in the same class?

Reply:
Services are basically global functions which can be reused.
I don't think these methods you created can be reused in any other case.
At least for now.

Reply:
Okay. Have removed the service.